### PR TITLE
Announcer  api types

### PR DESCRIPTION
**Pull request by mdittmer**
## Description

Test and implement HTTP API request/response golang types and conversion functions between internal API types.

### CODE REVIEW

File: revisions/api/types.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+	if len(rs) < len(epochs) {
+		return latest, errMissingRevision
+	}
+
+	return latest, nil
+}
+
+// EpochsRequest models a request for the epochs supported by the service.
+type EpochsRequest struct{}
```
Comment (lukebjerring) [inline revisions/api/types.go]: What's this used by?

File: revisions/api/types.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+// EpochsResponse models a response for the epochs supported by the service.
+type EpochsResponse []Epoch
+
+// RevisionsRequest models a request for the announced revisions.
+type RevisionsRequest struct {
+	Epochs       []epoch.Epoch `json:"epochs,omitempty"`
+	NumRevisions int           `json:"num_revisions,omitempty"`
+	Now          time.Time     `json:"now,omitempty"`
```
Comment (lukebjerring) [inline revisions/api/types.go]: I don't know about `Now` as a field passed by the client. Maybe `Time`, `From` or `At` ?

File: revisions/api/types.go (hunk 1 above)
Comment (mdittmer) [reply revisions/api/types.go]: Nothing. It's leftover from when I was playing with abstract validation of requests and needed a struct for params of all request types.

File: revisions/api/types.go (hunk 2 above)
Comment (mdittmer) [reply revisions/api/types.go]: Changed to `At` (along with announcer layer field by the same name).

File: revisions/api/types.go (hunk 1 above)
Comment (mdittmer) [reply revisions/api/types.go]: Deleted.

**Pull request merged**